### DISCUSSION TITLE
Do tools work with components using new vue setup syntax?

### DISCUSSION THREAD

**Discussion by jquirogaa** [laravel/nova-issues#5494] [Q&A] [answered]
I always got this warning and never load my components when using that new syntax, using onMounted hook as compotion api:

```
<script setup>
import { onMounted, ref } from "vue";
const count = ref(0);
 onMounted(() =>console.log(count.value++));
</script>
```
![image](https://user-images.githubusercontent.com/95649309/233810309-595df1f8-8669-479b-807b-29fcdb5a0d83.png)

when using mounted as optional api works fine, I just wanted to use that syntax I think it looks clean

**Comment by crynobone** [comment, marked as answer]
https://github.com/laravel/nova-dusk-suite/blob/master/nova-components/SidebarTool/resources/js/pages/Tool.vue

It works fine, some of Nova's internal component use composition API too.
